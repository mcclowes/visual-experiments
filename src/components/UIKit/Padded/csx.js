import styled from "styled-components/macro";

export const PaddedWrapper = styled.div`
  ${props => (props.space === "p1" ? "padding: 0.125em;" : "")};
  ${props => (props.space === "p2" ? "padding: 0.25em;" : "")};
  ${props => (props.space === "p3" ? "padding: 0.5em;" : "")};
  ${props => (props.space === "p4" ? "padding: 0.75em;" : "")};
  ${props => (props.space === "p5" ? "padding: 1em;" : "")};

  ${props =>
    props.space === "py1"
      ? "padding-top: 0.125em; padding-bottom: 0.125em;"
      : ""};
  ${props =>
    props.space === "py2"
      ? "padding-top: 0.25em; padding-bottom: 0.25em;"
      : ""};
  ${props =>
    props.space === "py3" ? "padding-top: 0.5em; padding-bottom: 0.5em;" : ""};
  ${props =>
    props.space === "py4"
      ? "padding-top: 0.75em; padding-bottom: 0.75em;"
      : ""};
  ${props =>
    props.space === "py5" ? "padding-top: 1em; padding-bottom: 1em;" : ""};

  ${props =>
    props.space === "px1"
      ? "padding-left: 0.125em; padding-right: 0.125em;"
      : ""};
  ${props =>
    props.space === "px2"
      ? "padding-left: 0.25em; padding-right: 0.25em;"
      : ""};
  ${props =>
    props.space === "px3" ? "padding-left: 0.5em; padding-right: 0.5em;" : ""};
  ${props =>
    props.space === "px4"
      ? "padding-left: 0.75em; padding-right: 0.75em;"
      : ""};
  ${props =>
    props.space === "px5" ? "padding-left: 1em; padding-right: 1em;" : ""};
`;
