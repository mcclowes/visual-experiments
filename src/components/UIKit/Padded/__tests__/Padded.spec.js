import Padded from "../Padded";
import React from "react";
import { shallow } from "enzyme";

const defaultProps = {};

const renderComponent = newProps => {
  const props = {
    ...defaultProps,
    ...newProps
  };

  return shallow(<Padded {...props} />);
};

let wrapper;

describe("Padded", () => {
  it("renders default component", () => {
    wrapper = renderComponent();

    expect(wrapper).toMatchSnapshot();
  });
});
