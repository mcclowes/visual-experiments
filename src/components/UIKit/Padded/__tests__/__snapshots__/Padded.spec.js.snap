// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Padded renders default component 1`] = `
<csx__PaddedWrapper
  space="p1"
/>
`;
